### ISSUE TITLE
Demo pipeline run fails with invalid memory address error

### ISSUE THREAD

**Issue by sameerbhadouria** [kubeflow/pipelines#4668]
### What steps did you take:
[A clear and concise description of what the bug is.]
Install Kubeflow with Pipelines configured in multi user isolation mode.
Create an experiment, select pre-installed demo pipeline (iris Classification), create a new run and fill in the Run Parameters and click Start.

### What happened:
The pipeline execution fails immediately with a upstream connect error. Following error is logged in the `ml-pipeine-api-server` 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x138 pc=0x11be2d6]

goroutine 405 [running]:
github.com/kubeflow/pipelines/backend/src/common/util.(*Workflow).VerifyParameters(0xc000d28408, 0xc000d2c1e0, 0x4, 0xc000d2c1e0)
	backend/src/common/util/workflow.go:66 +0x96
github.com/kubeflow/pipelines/backend/src/apiserver/resource.(*ResourceManager).CreateRun(0xc0001601c0, 0xc000b3fb20, 0xc000c9b050, 0xc000cb2b30, 0x2)
	backend/src/apiserver/resource/resource_manager.go:326 +0x20e
github.com/kubeflow/pipelines/backend/src/apiserver/server.(*RunServer).CreateRun(0xc000344328, 0x196e1e0, 0xc000c9b050, 0xc000c9b080, 0xc000344328, 0x1, 0x1)
	backend/src/apiserver/server/run_server.go:43 +0xc5
github.com/kubeflow/pipelines/backend/api/go_client._RunService_CreateRun_Handler.func1(0x196e1e0, 0xc000c9b050, 0x16b4d60, 0xc000c9b080, 0x1, 0x0, 0x25d5a20, 0x0)
	bazel-out/k8-opt/bin/backend/api/linux_amd64_stripped/go_client_go_proto%/github.com/kubeflow/pipelines/backend/api/go_client/run.pb.go:1277 +0x86
main.apiServerInterceptor(0x196e1e0, 0xc000c9b050, 0x16b4d60, 0xc000c9b080, 0xc000c98fe0, 0xc000c99000, 0x15c1f00, 0x25f3a90, 0x173b520, 0xc000b67b00)
	backend/src/apiserver/interceptor.go:30 +0xf4
github.com/kubeflow/pipelines/backend/api/go_client._RunService_CreateRun_Handler(0x16e7b80, 0xc000344328, 0x196e1e0, 0xc000c9b050, 0xc000c9cfa0, 0x182d9b0, 0x0, 0x0, 0xc000a61c80, 0x16c)
	bazel-out/k8-opt/bin/backend/api/linux_amd64_stripped/go_client_go_proto%/github.com/kubeflow/pipelines/backend/api/go_client/run.pb.go:1279 +0x158
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000346780, 0x197b2c0, 0xc00094e780, 0xc000b67b00, 0xc000868270, 0x25650c0, 0x0, 0x0, 0x0)
	external/org_golang_google_grpc/server.go:966 +0x4a2
google.golang.org/grpc.(*Server).handleStream(0xc000346780, 0x197b2c0, 0xc00094e780, 0xc000b67b00, 0x0)
	external/org_golang_google_grpc/server.go:1245 +0xd61
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc00047aa00, 0xc000346780, 0x197b2c0, 0xc00094e780, 0xc000b67b00)
	external/org_golang_google_grpc/server.go:685 +0x9f
created by google.golang.org/grpc.(*Server).serveStreams.func1
	external/org_golang_google_grpc/server.go:683 +0xa1
```
Here is the HTTP request that goes from the UI as it looks like some error in validation:
```
{
  "description":"",
  "name":"Run of [Demo] TFX - Iris classification pipeline (c7eef)",
  "pipeline_spec":{
    "parameters":[
      {
        "name":"pipeline-root",
        "value":"gs:\/\/<my bucket>\/tfx_iris\/{{workflow.uid}}"
      },
      {
        "name":"data-root",
        "value":"gs:\/\/ml-pipeline\/sample-data\/iris\/data"
      },
      {
        "name":"module-file",
        "value":"\/tfx-src\/tfx\/examples\/iris\/iris_utils_native_keras.py"
      }
    ]
  },
  "resource_references":[
    {
      "key":{
        "id":"e562d936-8982-45a6-8dcb-ddcdbd06e1da",
        "type":"EXPERIMENT"
      },
      "relationship":"OWNER"
    },
    {
      "key":{
        "id":"280aa4ec-27b0-42d5-9bb1-0ce066eae51b",
        "type":"PIPELINE_VERSION"
      },
      "relationship":"CREATOR"
    }
  ],
  "service_account":""
}
```
### What did you expect to happen:
Succesful run.

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?

As part of the Kubeflow installation

KFP version: 
v1.0.0

KFP SDK version: 

### Anything else you would like to add:
[Miscellaneous information that will assist in solving the issue.]
I am trying to set up kubeflow with CloudSQL and the Minio with GCS gateway instead of using the default persistent disk based approach. Kubeflow pipelines supports this although Kubeflow top level project does not officially supports it.

With the Minio using gcs I had to also patch the `pipeline-upstream-install-config` ConfigMap to update the bucketname from default `mlpipeline` to custom bucket that I am using. This bucket is also specified as an input in the run parameters. Without the ConfigMap patch, the Pipeline starts, reads data, does some processing but fails to write that bucket since buckets are global.

/kind bug

backend

**Comment by Bobgy** [comment]
I think this is
duplicate of https://github.com/kubeflow/pipelines/issues/4389.

And it's already been fixed and released in KFP 1.0.1 https://github.com/kubeflow/pipelines/blob/1.0.4/CHANGELOG.md#101-2020-09-14

**Comment by sameerbhadouria** [comment]
@Bobgy Thanks for the follow up. Just want to make sure this isn't related to using Minio with GCS gateway and an external CloudSQL MySQL instance. I was able to run the pipelines in a previous installation of Kubeflow pipeline, but with the internal Mysql instance and Minio both backed by a persistent disk.
